### PR TITLE
I can delete from logstream

### LINKED ISSUE #2304: I can delete data from the logstream
AT:
* I can delete records from the logstream/log storage which position is lower than a given number
* the data is deleted from the file system
* if the position is inside a log segment then it deletes only previous segments (i.e. don't delete data within a segment)

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #2304

### CODE REVIEW

**Comment by menski** [comment]
I rebased this branch with the latest develop to pull in the cloud CI config.

**Comment by Zelldon** [comment]
@menski thanks

File: logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java [hunk 1]
```diff
@@ -207,6 +212,139 @@ public void shouldNotDeleteOnCloseByDefault() {
...
+    final byte[] largeEvent = new byte[remainingCapacity];
+
+    // written from segment 0 4096 -> 8192, idx block address 4096
+    final long firstPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+    // written from segment 1 4096 -> 8192, but idx block address segment 0 - 8192
+    final long secondPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+    // written from segment 2 4096 -> 8192, but idx block address segment 1 - 8192
+    final long thirdPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
+    // written from segment 3 4096 -> 8192, but idx block address segment 2 - 8192
+    final long fourthPosition = writeEvent(logStream, BufferUtil.wrapArray(largeEvent));
```
Comment (saig0) [inline logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java]: The test setup is similar in all 3 cases. Please extract it.

File: logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java [hunk 2]
```diff
@@ -207,6 +212,139 @@ public void shouldNotDeleteOnCloseByDefault() {
...
+
+    logStream.setCommitPosition(fourthPosition);
+    waitUntil(() -> logStream.getLogBlockIndex().getLastPosition() >= fourthPosition);
+
+    // when
+    logStream.delete(fourthPosition);
+
+    // then
+    assertThat(events(logStream).count()).isEqualTo(2);
```
Comment (saig0) [inline logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java]: It is not clear why we delete only 2 of 4 segments and not 3. Please add a comment or change the test case.

**Review by saig0 — CHANGES_REQUESTED** [review]
Just some small things :)

File: logstreams/src/main/java/io/zeebe/logstreams/impl/log/fs/FsLogSegments.java [hunk 3]
```diff
@@ -39,6 +39,17 @@ public void addSegment(FsLogSegment segment) {
     this.segmentCount = newSegments.length; // volatile store
   }
 
+  public void removeSegments(int segmentId) {
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply logstreams/src/main/java/io/zeebe/logstreams/impl/log/fs/FsLogSegments.java]: Your renamings are evil.

[the remark this answers is not in the document]
Comment (Zelldon) [reply logstreams/src/main/java/io/zeebe/logstreams/impl/log/fs/FsLogSegments.java]: We have an check in the FsLogStorage, so I will then just ignore it if it is greater?

**Comment by Zelldon** [comment]
Please see my latest changes

**Review by saig0 — APPROVED** [review]

**Pull request merged**